server/src/test/java/org/opensearch/action/admin/indices/forcemerge/ForceMergeRequestTests.java

```diff
@@ -97,23 +97,30 @@ public void testSerialization() throws Exception {
     public void testBwcSerialization() throws Exception {
         {
             final ForceMergeRequest sample = randomRequest();
-            final Version compatibleVersion = VersionUtils.randomCompatibleVersion(random(), Version.CURRENT);
+            final Version version = VersionUtils.randomCompatibleVersion(random(), Version.CURRENT);
             try (BytesStreamOutput out = new BytesStreamOutput()) {
-                out.setVersion(compatibleVersion);
+                out.setVersion(version);
                 sample.writeTo(out);
 
                 try (StreamInput in = out.bytes().streamInput()) {
-                    in.setVersion(Version.CURRENT);
+                    in.setVersion(version);
                     TaskId.readFromStream(in);
                     in.readStringArray();
                     IndicesOptions.readIndicesOptions(in);
                     int maxNumSegments = in.readInt();
                     boolean onlyExpungeDeletes = in.readBoolean();
                     boolean flush = in.readBoolean();
+                    boolean primaryOnly = false;
+                    if (version.onOrAfter(Version.V_2_13_0)) {
+                        primaryOnly = in.readBoolean();
+                    }
                     String forceMergeUUID = in.readOptionalString();
                     assertEquals(sample.maxNumSegments(), maxNumSegments);
                     assertEquals(sample.onlyExpungeDeletes(), onlyExpungeDeletes);
                     assertEquals(sample.flush(), flush);
+                    if (version.onOrAfter(Version.V_2_13_0)) {
```

Comment:
you don't need this check as in 2.13 version we expect the default value of `false` for `primaryOnly` field. So if version is < 2.13, we will expect that default value otherwise whatever was serialized.

Reply:
But we've set `primaryOnly` to a random boolean: https://github.com/opensearch-project/OpenSearch/blob/0560d9996a182bc494151e97b0d0a23b7775ba19/server/src/test/java/org/opensearch/action/admin/indices/forcemerge/ForceMergeRequestTests.java#L163, so this check is needed.